benefits/oauth/hooks.py

```diff
@@ -0,0 +1,20 @@
+from cdt_identity.hooks import DefaultHooks
+from django.shortcuts import redirect
+import sentry_sdk
+
+from benefits.routes import routes
+from . import analytics
+
+
+class OAuthHooks(DefaultHooks):
+    @classmethod
+    def pre_login(cls, request):
+        super().pre_login(request)
+        analytics.started_sign_in(request)
+
+    @classmethod
+    def system_error(cls, request, exception):
+        super().system_error(request, exception)
+        analytics.error(request, message=str(exception), operation="authorize_redirect")
```

Comment:
@thekaveman I noticed we have this `operation` parameter in the analytics call, and the value is different depending on when the system error occurred.

The 3 different values I see are:
- [`"init"`](https://github.com/cal-itp/benefits/blob/main/benefits/oauth/views.py#L40)
- [`"authorize_redirect"`](https://github.com/cal-itp/benefits/blob/main/benefits/oauth/views.py#L81)
- [`"authorize_access_token"`](https://github.com/cal-itp/benefits/blob/main/benefits/oauth/views.py#L116)

I think we need to modify this hook slightly to handle that, and I have some ideas on how it could change, but I'm curious if you have any immediate thoughts?

Reply:
The easiest thing to do I think is to have the hook take an `operation` parameter, and `django-cdt-identity` would call the `system_error` hook with the correct value for `operation` at the appropriate places. It's pretty specific to this analytics use case, but maybe that's ok.

Reply:
Hmmm good catch, and I like your straightforward approach 🙌 The operations seem to roughly correspond to the different methods the OAuth client class we use, so maybe we stick to that as our guide for what values to pass to `operation`.

Reply:
Thanks @thekaveman. https://github.com/Office-of-Digital-Services/django-cdt-identity/pull/49 is ready for review